src/Components/Aspire.Azure.Messaging.EventHubs/README.md

```diff
@@ -0,0 +1,182 @@
+# Aspire.Azure.Messaging.EventHubs
+
+Offers options for registering an [EventHubProducerClient](https://learn.microsoft.com/en-us/dotnet/api/azure.messaging.eventhubs.producer.eventhubproducerclient), an [EventHubConsumerClient](https://learn.microsoft.com/dotnet/api/azure.messaging.eventhubs.consumer.eventhubconsumerclient), an [EventProcessorClient](https://learn.microsoft.com/dotnet/api/azure.messaging.eventhubs.eventprocessorclient) or a [PartitionReceiver](https://learn.microsoft.com/en-us/dotnet/api/azure.messaging.eventhubs.primitives.partitionreceiver) in the DI container for connecting to Azure Event Hubs.
+
+## Getting started
+
+### Prerequisites
+
+- Azure subscription - [create one for free](https://azure.microsoft.com/free/)
+- Azure Event Hubs namespace, learn more about how to [add an Event Hubs namespace](https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-create). Alternatively, you can use a connection string, which is not recommended in production environments.
+
+### Install the package
+
+Install the .NET Aspire Azure Event Hubs library with [NuGet](https://www.nuget.org):
+
+```dotnetcli
+dotnet add package Aspire.Azure.Messaging.EventHubs
+```
+
+## Supported clients with Options classes
+
+The following clients are supported by the library, along with their corresponding Options classes:
+
+| Client Type [index]              | Options Class                 |
+|------------------------------------------|-------------------------------|
+| EventHubProducerClient[1] | EventHubProducerClientOptions |
+| EventHubConsumerClient[2] | EventHubConsumerClientOptions |
+| EventProcessorClient[3]   | EventProcessorClientOptions   |
+| PartitionReceiver[4]      | PartitionReceiverOptions      |
+
+### AzureMessagingEventHubsSettings shared configuration
+
+The shared settings for all clients are suffixed below with one or more
+indices corresponding to the client types listed above that may use them:
+
+| Member                                                                                   | Type             |
+|------------------------------------------------------------------------------------------|------------------|
+| ConnectionString[1,2,3,4]   | string?          |
+| Namespace[1,2,3,4]          | string?          |
+| EventHubName[1,2,3,4]       | string?          |
+| ConsumerGroup[2,3,4]        | string?          |
+| BlobClientConnectionName[3] | string?          |
+| BlobContainerName[3]        | string?          |
+| PartitionId[4]              | string?          |
+| EventPosition[4]            | EventPosition    |
+| Credential[1,2,3,4]         | TokenCredential? |
+| Tracing[1,2,3,4]            | bool             |
```

Comment:
Would it make more sense to split these into separate classes? That way we didn't have properties on Settings types that didn't work?

For example, I could imagine this hierarchy:

```C#
abstract class AzureMessagingEventHubsBaseSettings
{
  ConnectionString
  Namespace
  EventHubName
  Credential
  Tracing
}

sealed class AzureMessagingEventHubsProducerSettings : AzureMessagingEventHubsBaseSettings {}

abstract class AzureMessagingEventHubsConsumerBaseSettings: AzureMessagingEventHubsBaseSettings
{
  ConsumerGroup
}

sealed class AzureMessagingEventHubsConsumerSettings : AzureMessagingEventHubsBaseSettings {}

sealed class AzureMessagingEventHubsProcessorSettings : AzureMessagingEventHubsBaseSettings
{
  BlobClientConnectionName
  BlobContainerName
}

sealed class AzureMessagingEventHubsPartitionReceiverSettings : AzureMessagingEventHubsBaseSettings
{
  PartitionId
  EVentPosition
}
```

Reply:
Absolutely makes sense from an objective, clean separation of concerns point of view. But from an Aspire point of view where it would mean delving into the Aspire.Hosting common framework to modify it to handle multiple types of settings classes, I don't know if this would be in scope for this PR. A reminder that this is the only componenty (Event Hubs) that has multiple client types; all others have a single client type, i.e. for ASB, it only has ServiceBusSender. This is the only way I could see to make this work within the p5 timeline. Feel free to bring this up internally with @davidfowl and @mitchdenny

Reply:
> from an Aspire point of view where it would seem to mean (afaict) delving into the Aspire.Hosting common framework to modify it to handle multiple types of settings classes

I don't follow this comment. "Aspire.Hosting" doesn't matter here. Each public API in the component can take its own Settings type:

``` 
public static void AddAzureEventProcessorClient(
     this IHostApplicationBuilder builder,
     string connectionName,
     Action<AzureMessagingEventHubsProcessorSettings>? configureSettings = null

...

    public static void AddAzureEventHubProducerClient(
        this IHostApplicationBuilder builder,
        string connectionName,
        Action<AzureMessagingEventHubsProducerSettings>? configureSettings = null
```

Reply:
Ah, I see -- thanks for the tip. Apologies -- I've only had my head in this code for the best part of a long weekend.

Reply:
@eerhardt Is there a way to flag the base settings types so they don't get picked up by the ConfigurationSchema source generator? Or does this even matter?

Reply:
Never mind -- I assume it will ignore abstract types.

Reply:
In general the ConfigurationSchema generator will skip the properties/types that don't work with the config binder source generator. There is a way to make the schema generator skip properties if you need to manually say to skip them:

https://github.com/dotnet/aspire/blob/23104ee048dcf99bc1bdcb5ff8ad01ee85921d3b/src/Components/Common/ConfigurationSchemaAttributes.cs#L29-L32

Reply:
Done.